lib/puppet/provider/package/pip.rb

```diff
@@ -31,7 +31,11 @@ def self.instances
     packages = []
     pip_cmd = self.pip_cmd
     return [] unless pip_cmd
-    execpipe "#{pip_cmd} freeze" do |process|
+    command = [pip_cmd, 'freeze']
+    if Puppet::Util::Package.versioncmp(self.pip_version, '8.1.0') >= 0 # a >= b
+        command << '--all'
```

Comment:
I rebased on master, resolved the conflict and updated my commit to reflect the same changes as #5903.